### PR TITLE
Prevent duplicates in restrictedFuncs

### PR DESCRIPTION
**Pull request by tenntenn**
## Summary
- avoid adding the same function multiple times
- add unit test ensuring duplicate names are deduplicated

### CODE REVIEW

File: restrictedfuncs_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package called
```
Comment (tenntenn) [inline restrictedfuncs_test.go:1]: called_test
